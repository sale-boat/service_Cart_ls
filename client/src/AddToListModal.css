.addToList{
    padding-top: 15px;
}

.addButton{
    width:100%;
    padding: 10px;
    border-radius: 3px;
    background: linear-gradient(to bottom,#f7f8fa,#e7e9ec);
    border-color: #ADB1B8 #A2A6AC #8D9096;
    border-width: 1px;
    text-align: left;
    font-size: 13px;
}


.list{
    display: none;
    position: fixed;
    z-index: 1;
    padding-top: 150px;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%; 
    overflow: auto;
    background-color: rgba(0,0,0,0.7);
}

.listHeader{
    text-align: left;
    background: linear-gradient(to bottom,#f7f7f7,#eaeaea);
    padding:10px 13px;
    line-height: 1.9;
    font-weight: bold;
    display : block;
    color: #111;
    font-size: 13px;
    margin: auto;
    width: 40%;
    border: 1px solid #e1e1e1;
    z-index: 100;
    border-radius: 4px 4px 0 0;
    position: relative;
}

.listHeaderText{
    color: #303030;
}

.exitButton{
    position: absolute;
    right: 5px;
    top: 5px;
    border-radius: 3px;
    width: 30px;
    height: 30px;
    display: inline-block;
    font-size: 15px;
    background: linear-gradient(to bottom,#f7f7f7,#eaeaea);
    text-align: center;
}

.listContent{
    background-color: #fff;
    padding:10px 14px;
    display: block;
    width: 40%;
    margin: auto;
}

.ownerText{
    display:block;
    font-weight: bold;
}

.ownerType{
    padding-left: 10px;
    width: 148px;
    height: 29px;
    -webkit-appearance: none;
    background: url(https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonarrow.png);
    background-repeat: no-repeat;
    background-position: right center;
    background-color: #f1f2f5;
}

.choose{
    border-top: 1px solid #e1e1e1;
    border-bottom: 1px solid #e1e1e1;
    padding-top: 10px;
    margin-top: 10px;
    padding-bottom: 25px;
}

.chooseText{
    color: #1e1e1e;
    font-weight: bold;
}


.listTypes{
    margin-top: 5px;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.darkCircle{
    width: 15px;
    height: 15px;
    padding-top:4px;
}
/* input[type="radio"]:after{
    width: 15px;
    height: 15px;
    top: -2px;
    left:-5px;
    border-radius: 15px;
    position: relative;
    background-color: white;
    content: '';
    display:inline-block;
    border: 1px solid black;
    visibility:visible;
}

input[type="radio"]:checked:after{
    width: 10px;
    height: 10px;
    top: -2px;
    left:-2px;
    border-radius: 8px;
    position:relative;
    background-color: orange;
    content: '';
    display:inline-block;
    visibility:visible;
} */



.shoppingList{
    border: 1px #ddd solid;
    height: 100px;
    width: 200px;
    border-radius:3px;
    margin-right: 10px;
    padding: 10px;
}

.wishList{
    border: 1px #ddd solid;
    height: 100px;
    width: 200px;
    border-radius:4px;
    margin-right: 10px;
    padding: 10px;
}

.ideaList{
    border: 1px #ddd solid;
    height: 100px;
    width: 200px;
    border-radius:3px;
    padding: 10px;
}

.shoppingListText, .wishListText, .ideaListText{
    color: #111;
    display: inline; 
}



.shoppingListText2, .wishListText2, .ideaListText2 {
    color: #76767E;
    padding-left: 20px;
    font-size: 12px;
}


.listNamesSection{
    margin-top: 20px;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
}

.listName{
    width: 300px;
    margin-right: 20px;
    padding: 15px;
}

.listNameText{
   color: #111;
   display: block;
   font-weight: bold;
}

.listNameInput{
    border-radius: 4px;
    height: 20px;
    width: 100%;
    margin-bottom: 15px;
}


.privacyText{
    color: #111;
    font-weight: bold;
 }

.privacy{
    width: 300px;
    padding-top: 15px;
    padding-left: 15px;
}


.private{
    width: 64px;
    height: 29px;
    border: 1px solid #d4d6d9;
    border-radius: 4px;
    margin-right: 5px;
    background-color: #fef8f2;
    border: 1px solid #f7cca6;
    font-weight: bold;
}

.public{
    width: 64px;
    height: 29px;
    border: 1px solid #d4d6d9;
    border-radius: 4px;
}

.personal{
    display: block;
    color: #acacac;
}

.listFooter{
    text-align: right;
    background: linear-gradient(to bottom,#f7f7f7,#eaeaea);
    width: 40%;
    border-radius: 0 0 4px 4px;
    margin: auto;
    display: block;
    padding:18px 14px 35px 14px;
}

.createList{
    background: linear-gradient(to bottom,#f7dfa5,#f0c14b);
    width: 83px;
    height: 29px;
    border-radius: 4px;
    border-color: #a88734 #9c7e31 #846a29;
    font-size: 10px;
}

.cancel{
    width: 60px;
    height: 29px;
    background: linear-gradient(to bottom,#f3f3f3,#fff);
    border-radius: 4px;
    margin-right: 5px;
    border-color: #ADB1B8 #A2A6AC #8D9096;
    font-size: 10px;
}