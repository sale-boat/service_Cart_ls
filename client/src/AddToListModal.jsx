import React from 'react';
import addToModal from './AddToListModal.css';

class AddToListModal extends React.Component {
  constructor(props) {
    super(props);
    this.addToListModalShow = this.addToListModalShow.bind(this);
    this.addToListModalHide = this.addToListModalHide.bind(this);
    this.hideButtonBorder = this.hideButtonBorder.bind(this);
    this.state = {
      addToListModalShow: false,
      buttonToExit: true,
    };
  }

  addToListModalShow() {
    this.setState({ addToListModalShow: true });
    const { buttonToExit } = this.state;
    if (buttonToExit === false) {
      this.setState({ buttonToExit: true });
    }
  }

  addToListModalHide(event) {
    console.log(event.target);
    if (event.target.className === addToModal.list || event.target.className === addToModal.exitButton) {
      this.setState({ addToListModalShow: false });
    }
  }

  hideButtonBorder() {
    this.setState({ buttonToExit: false });
  }

  render() {
    const { addToListModalShow } = this.state;
    const style = { display: addToListModalShow ? 'block' : 'none' };
    const { buttonToExit } = this.state;
    const style1 = {
      border: buttonToExit ? '1px solid orange' : 'transparent',
      boxShadow: buttonToExit ? '0 0 3px #ff8000' : ''
    };

    return (
      <div className={addToModal.addToList}>
        <div className={addToModal.list} style={style} onClick={this.addToListModalHide} >
          <div className={addToModal.listHeader} onClick={this.hideButtonBorder}>
            <span className={addToModal.listHeaderText}>Add to your list</span>
            <button type="button" value="x" className={addToModal.exitButton} onClick={this.addToListModalHide} style={style1}>x</button>
          </div>
          <div className={addToModal.listContent} onClick={this.hideButtonBorder}>
            <div className={addToModal.listOwner}>
              <span className={addToModal.ownerText}>This list is for</span>
              <select className={addToModal.ownerType}>
                <option>You</option>
                <option>Someone else</option>
              </select>
            </div>
            <div className={addToModal.choose}>
              <span className={addToModal.chooseText}>Choose a list type</span>
              <div className={addToModal.listTypes}>
                <div className={addToModal.shoppingList}>
                  <div className={addToModal.shoppingListInner}>
                    <input type="radio" name="radio" />
                    {' '}
                    <h3 className={addToModal.shoppingListText}>Shopping List</h3>
                    <br />
                    <div className={addToModal.shoppingListText2}>
                      <span>Add items you want to shop for.</span>
                    </div>
                  </div>
                </div>
                <div className={addToModal.wishList}>
                  <div className={addToModal.wishListInner}>
                    <input type="radio" name="radio" />
                    {' '}
                    <h3 className={addToModal.wishListText}>Wish List</h3>
                    <br />
                    <div className={addToModal.wishListText2}>
                      <span>Let people know what gifts you'd like.</span>
                    </div>
                  </div>
                </div>
                <div className={addToModal.ideaList}>
                  <div className={addToModal.ideaListInner}>
                    <input type="radio" name="radio" />
                    {' '}
                    <h3 className={addToModal.ideaListText}>Idea List</h3>
                    <br />
                    <div className={addToModal.ideaListText2}>
                      <span>Make a list for the Amazon community.</span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className={addToModal.listNamesSection}>
              <div className={addToModal.listName}>
                <span className={addToModal.listNameText}>List name</span>
                <input type="text" value="Shopping List" className={addToModal.listNameInput} />
                <input type="checkbox" />
                Keep purchased items on your list
              </div>
              <div className={addToModal.privacy}>
                <span className={addToModal.privacyText}>Privacy</span>
                <br />
                <button type="button" value="Private" className={addToModal.private}>Private</button>
                <button type="button" value="Public" className={addToModal.public}>Public</button>
                <span className={addToModal.personal}>Only you can see this list.</span>
              </div>
            </div>
          </div>
          <div className={addToModal.listFooter} onClick={this.hideButtonBorder}>
            <button type="button" className={addToModal.cancel}>Cancel</button>
            <button type="button" className={addToModal.createList}>Create List</button>
          </div>
        </div>
        <input type="submit" value="Add To List" className={addToModal.addButton} onClick={this.addToListModalShow} />
      </div>
    );
  }
}

export default AddToListModal;
