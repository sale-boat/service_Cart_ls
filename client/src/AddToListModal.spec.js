import React from 'react';
import { mount, shallow } from 'enzyme';
import AddToListModal from './AddToListModal.jsx';

describe('AddToList Component', () => {
  it('should render', () => {
    const addToList = shallow(<AddToListModal />);
    expect(addToList).toMatchSnapshot();
  });

  it('should call add to list modal show when clicked and modal should exist', () => {
    const addToListModalShow = jest.spyOn(AddToListModal.prototype, 'addToListModalShow');
    const wrapper = mount(<AddToListModal /> );
    const addButton = wrapper.find('.addButton');
    addButton.simulate('click');
    expect(addToListModalShow).toHaveBeenCalledTimes(1);
    expect(wrapper.find('.list').exists()).toEqual(true);
  });

  it('should set the addToListModalShow state to true when Add To List is clicked', () => {
    const wrapper = shallow(<AddToListModal />);
    const addButton = wrapper.find('.addButton');
    addButton.simulate('click');
    expect(wrapper.state().addToListModalShow).toEqual(true);
  })
 
});