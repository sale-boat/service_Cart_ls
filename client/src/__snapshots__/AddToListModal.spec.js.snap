// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToList Component should render 1`] = `
<div
  className="addToList"
>
  <div
    className="list"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="list-header"
      onClick={[Function]}
    >
      <span
        className="list-header-text"
      >
        Add to your list
      </span>
      <button
        className="exitButton"
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid orange",
            "boxShadow": "0 0 3px #ff8000",
          }
        }
        type="button"
        value="x"
      >
        x
      </button>
    </div>
    <div
      className="list-content"
      onClick={[Function]}
    >
      <div
        className="listOwner"
      >
        <span
          className="ownerText"
        >
          This list is for
        </span>
        <select
          className="ownerType"
        >
          <option>
            You
          </option>
          <option>
            Someone else
          </option>
        </select>
      </div>
      <div
        className="choose"
      >
        <span
          className="chooseText"
        >
          Choose a list type
        </span>
        <div
          className="list-types"
        >
          <div
            className="shopping-list"
          >
            <div
              className="shopping-list-inner"
            >
              <input
                name="radio"
                type="radio"
              />
               
              <h3
                className="shopping-list-text"
              >
                Shopping List
              </h3>
              <br />
              <div
                className="shopping-list-text2"
              >
                <span>
                  Add items you want to shop for.
                </span>
              </div>
            </div>
          </div>
          <div
            className="wish-list"
          >
            <div
              className="wish-list-inner"
            >
              <input
                name="radio"
                type="radio"
              />
               
              <h3
                className="wish-list-text"
              >
                Wish List
              </h3>
              <br />
              <div
                className="wish-list-text2"
              >
                <span>
                  Let people know what gifts you'd like.
                </span>
              </div>
            </div>
          </div>
          <div
            className="idea-list"
          >
            <div
              className="idea-list-inner"
            >
              <input
                name="radio"
                type="radio"
              />
               
              <h3
                className="idea-list-text"
              >
                Idea List
              </h3>
              <br />
              <div
                className="idea-list-text2"
              >
                <span>
                  Make a list for the Amazon community.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="list-names-section"
      >
        <div
          className="list-name"
        >
          <span
            className="list-name-text"
          >
            List name
          </span>
          <input
            className="list-name-input"
            type="text"
            value="Shopping List"
          />
          <input
            type="checkbox"
          />
          Keep purchased items on your list
        </div>
        <div
          className="privacy"
        >
          <span
            className="privacy-text"
          >
            Privacy
          </span>
          <br />
          <button
            className="private"
            type="button"
            value="Private"
          >
            Private
          </button>
          <button
            className="public"
            type="button"
            value="Public"
          >
            Public
          </button>
          <span
            className="personal"
          >
            Only you can see this list.
          </span>
        </div>
      </div>
    </div>
    <div
      className="list-footer"
      onClick={[Function]}
    >
      <button
        className="cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="createList"
        type="button"
      >
        Create List
      </button>
    </div>
  </div>
  <input
    className="addButton"
    onClick={[Function]}
    type="submit"
    value="Add To List"
  />
</div>
`;
